lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -131,12 +176,13 @@ SmallVector<ComponentPortInfo> calyx::getComponentPortInfo(Operation *op) {
   auto component = dyn_cast<ComponentOp>(op);
   auto portTypes = getComponentType(component).getInputs();
   auto portNamesAttr = component.portNames();
-  uint64_t numInPorts = component.numInPorts();
+  auto portDirectionsAttr =
+      component->getAttrOfType<mlir::IntegerAttr>(direction::attrKey);
 
   SmallVector<ComponentPortInfo> results;
   for (uint64_t i = 0, e = portNamesAttr.size(); i != e; ++i) {
-    auto dir = i < numInPorts ? PortDirection::INPUT : PortDirection::OUTPUT;
-    results.push_back({portNamesAttr[i].cast<StringAttr>(), portTypes[i], dir});
+    results.push_back({portNamesAttr[i].cast<StringAttr>(), portTypes[i],
```

Comment:
Nit (feel free to ignore): curly braces not necessary if one-liner. However, there is quite a lot going on here :-).